plugin/trino-phoenix/src/main/java/io/trino/plugin/phoenix/PhoenixMetadata.java

```diff
@@ -83,17 +83,25 @@ public JdbcTableHandle getTableHandle(ConnectorSession session, SchemaTableName
     }
 
     @Override
-    public ConnectorTableMetadata getTableMetadata(ConnectorSession session, ConnectorTableHandle table)
+    public ColumnMetadata getColumnMetadata(ConnectorSession session, ConnectorTableHandle tableHandle, ColumnHandle columnHandle)
+    {
+        return getColumnMetadata((JdbcColumnHandle) columnHandle);
+    }
+
+    private ColumnMetadata getColumnMetadata(JdbcColumnHandle columnHandle)
     {
-        return getTableMetadata(session, table, false);
+        ColumnMetadata column = columnHandle.getColumnMetadata();
+        return ROWKEY.equalsIgnoreCase(column.getName())
+                ? ColumnMetadata.builderFrom(column).setHidden(true).build()
```

Comment:
Why not fix `columnHandle.getColumnMetadata()` so that it returns the correct thing for the ROWKEY as well?

Reply:
It will require implement new `PhoenixColumnHandle` class. 
Such functionality is not present in any of existing `ColumnHandle` classes.
Maybe this is a good idea for the next improvement (not this pr)

Reply:
Please use regular `if`, also indent builder
```
 ColumnMetadata.builderFrom(column)
      .setHidden(true)
      .build()
```

Reply:
done

Reply:
Other than that please see @findepi comment. The fact that column handle returns incorrect metadata is concerning.

Reply:
I'll take it into account and will consider ways to improve it, but not within this pr. Currently it is inline with other connectors which have built in columns.

Reply:
https://github.com/trinodb/trino/pull/6807#pullrequestreview-584185118
> Why not in this PR? Let's talk.

This requires `JdbcColumnHandle`->`PhoenixColumnHandle` migration and this is too huge change: `JdbcColumnHandle` is `final` and it makes me to anticipate huge code change and discussion around it.

Reply:
Thanks.

@findepi ?

Reply:
> This requires JdbcColumnHandle->PhoenixColumnHandle migration and this is too huge change: JdbcColumnHandle is final and it makes me to anticipate huge code change and discussion around it.

Did you consider adding `hidden` field to `JdbcColumnHandle`?

Reply:
@findepi, great idea! 
But still it does not help, because to achieve
> Why not fix columnHandle.getColumnMetadata() so that it returns the correct thing for the ROWKEY as well?

 this logic
```
    private ColumnMetadata getColumnMetadata(JdbcColumnHandle columnHandle)
    {
        ColumnMetadata column = columnHandle.getColumnMetadata();
        if (ROWKEY.equalsIgnoreCase(column.getName())) {
            return ColumnMetadata.builderFrom(column)
                    .setHidden(true)
                    .build();
        }
        return column;
    }
```
has to land to `ColumnHandle` class. And this is possible only with removing `final` modifier from `JdbcColumnHandle` and extending this class.

Reply:
The issue is with `io.trino.plugin.jdbc.BaseJdbcClient#getColumns(io.trino.spi.connector.ConnectorSession, io.trino.plugin.jdbc.JdbcTableHandle)` which is used in Phoenix, it returns all column with `rowkey`. That method is not capable to return hidden columns. So we would need to update column handle for this column there. And then I see that below should just work:
> Did you consider adding hidden field to JdbcColumnHandle?